### PR TITLE
fix: Register and run default preprocessor in child process

### LINKED ISSUE #8634: Preprocessor opens another Cypress instance when bundling
This issue happens in Cypress v5.0.0 and does not occur in v4.12.1

If you try to run Cypress tests in https://github.com/eggheadio/egghead-next then a curious thing happens: it pops a new Cypress instance while trying to bundle.

Recording: https://jumpshare.com/v/v5mfJi5QlSbsd7IpBFAb

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Closes #8634

### User facing changelog

- Fixed an issue where certain preprocessor-related plugins would cause tests not to run and a duplicate instance of Cypress to be spawned.

### Additional details

We run user-registered file preprocessors in the plugins child process, but when the user hasn't registered their own prepocessor, we register the default preprocessor in the main Cypress process. This causes [issues with preprocessor-related plugins](https://github.com/cypress-io/cypress/issues/8634#issuecomment-697874253) like Istanbul that use `process.execPath` to execute code because `process.execPath` is the Cypress binary in the main process and not `node`.

This fixes that by always registering and running the preprocessor in the child process, even our own default preprocessor.

Also, we used to not create the plugins child process if the user didn't have a plugins file. Now we always create it since it is necessary for running the default preprocessor.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- N/A Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- N/A Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- N/A Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/server/test/unit/plugins/child/run_plugins_spec.js [hunk 1]
```diff
@@ -15,6 +16,17 @@ const Fixtures = require(`${root}../../test/support/helpers/fixtures`)
...
+const deferred = () => {
+  let reject
+  let resolve
+  const promise = new Promise((_resolve, _reject) => {
+    resolve = _resolve
+    reject = _reject
+  })
+
+  return { promise, resolve, reject }
+}
```
Comment (flotwig) [inline packages/server/test/unit/plugins/child/run_plugins_spec.js:28]: could also use `Bluebird.defer`, it's deprecated but i won't tell :shushing_face:

**Review by flotwig — DISMISSED** [review]
LGTM

**Comment by jennifer-shehane** [comment]
This PR introduced a new issue: https://github.com/cypress-io/cypress/issues/8864

**Comment by jennifer-shehane** [comment]
This PR introduced this new issue: https://github.com/cypress-io/cypress/issues/9145

**Comment by jennifer-shehane** [comment]
This PR introduced a new issue: https://github.com/cypress-io/cypress/issues/14587

**Comment by jennifer-shehane** [comment]
This PR also introduced https://github.com/cypress-io/cypress/issues/9326

**Pull request merged**